### ISSUE TITLE
mxnet - caffe training difference

### ISSUE THREAD

**Issue by korneil** [dmlc/mxnet#680]
I have a folder of images with 4 classes (with equal images in each class). I create a record with im2rec for training and evaluation, then run train_imagenet.py from examples folder.

If e.g. I use googlenet as the network, training and validation accuracy is floating around 0.25 +- 0.5 even after 100 epoch.

If I train with caffe on the same jpg imageset, it get's ~99.99 accuracy after 12 or 13 epoch (I am using nvidia's digits for that).

I tried different nets, also a quite simple one but I always get accuracy around 0.25, showing that in case of a 4 class classification it's either random or always give the same prediction.

Do you have any idea what am I doing wrong?

**Comment by antinucleon** [comment]
Could you post your training script? Seems your forget shuffle data /. initialization /  too large learning rate, then lead the training fail.

**Comment by korneil** [comment]
When I generate the train.list and validation.lst I shuffle the values. Then I create the record files with im2rec (which I guess also shuffles), then I am using the standard train_imagenet.py from the examples folder using different networks (also googlenet).

**Comment by antinucleon** [comment]
For googlenet, the most important is initialization and current one may be inaccurate because I use Inception BN but not the old Googlenet. Batch norm will reduce the importance of init value. Could you test Inception Net with batch normalization? I will fix the init bug later.

**Comment by korneil** [comment]
With inception-bn it works well, thank you!

**Comment by antinucleon** [comment]
@Normal network training should be fixed in https://github.com/dmlc/mxnet/pull/713
